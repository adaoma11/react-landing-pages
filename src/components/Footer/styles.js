import styled, { css } from 'styled-components';

export const Footer = styled.footer`
  ${({ theme, $hasBg }) => css`
    display: flex;
    align-items: center;
    justify-content: center;
    height: 150px;
    background-color: ${$hasBg ? theme.colors.primary : theme.colors.white};
  `}
`;
