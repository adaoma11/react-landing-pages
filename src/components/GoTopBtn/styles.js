import styled, { css } from 'styled-components';

export const Link = styled.a`
  ${({ theme }) => css`
    display: flex;
    width: 40px;
    height: 40px;
    position: fixed;
    bottom: 2rem;
    right: 2rem;
    color: ${theme.colors.white};
    background-color: ${theme.colors.primary};
    border-radius: 5px;
    opacity: 0.8;
  `}
`;
