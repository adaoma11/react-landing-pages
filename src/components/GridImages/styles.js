import styled from 'styled-components';

export const GridContainer = styled.div`
  display: grid;
  grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
  width: 100%;
  gap: 20px;
  margin-top: 50px;
`;

export const GridItem = styled.div`
  overflow: hidden;
`;

export const Image = styled.img`
  width: 100%;
  transition: all 0.5s ease-in-out;

  &:hover {
    transform: scale(1.2) rotate(5deg);
  }
`;
