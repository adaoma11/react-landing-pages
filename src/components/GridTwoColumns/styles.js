import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${({ theme }) => css`
    display: grid;
    grid-template-columns: 1fr 1fr;
    align-items: center;
    gap: 30px;

    @media ${theme.medias.medium} {
      grid-template-columns: 1fr;
    }
  `}
`;

export const TextContainer = styled.div`
  display: flex;
  flex-direction: column;
  gap: 30px;
`;

export const Image = styled.div`
  ${({ $imgSrc }) => css`
    display: block;
    min-height: 500px;
    background-image: url(${$imgSrc});
    background-repeat: no-repeat;
    background-position: center;
    background-size: 60%;
  `}
`;
