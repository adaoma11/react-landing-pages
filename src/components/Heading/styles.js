import styled, { css } from 'styled-components';

export const Header = styled.header`
  margin-bottom: 30px;
`;

export const Title = styled.h1`
  ${({ theme, $dark, size, $upperCase, $align }) => css`
    color: ${$dark ? theme.colors.primary : theme.colors.white};
    font-size: ${theme.fonts.sizes[size]};
    margin-bottom: 1rem;
    text-transform: ${$upperCase ? 'uppercase' : 'none'};
    ${$align ? 'text-align: ' + $align : ''};

    ${size == 'xlarge' &&
    css`
      @media ${theme.medias.small} {
        font-size: ${theme.fonts.sizes.large};
      }
    `};
  `}
`;
