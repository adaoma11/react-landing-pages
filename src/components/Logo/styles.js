import styled, { css } from 'styled-components';

export const Container = styled.a`
  ${({ theme }) => css`
    font-family: ${theme.fonts.family.heading};
    font-size: ${theme.fonts.sizes.large};
    font-weight: bold;
    color: ${theme.colors.primary};
    text-decoration: none;

    > img {
      min-height: 40px;
      max-height: 50px;
      max-width: 150px;
      min-width: 40px;
    }
  `}
`;
