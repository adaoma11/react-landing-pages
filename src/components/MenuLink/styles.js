import styled, { css } from 'styled-components';

export const Link = styled.a`
  ${({ theme }) => css`
    display: block;
    padding: 15px;
    font-family: ${theme.fonts.family.default};
    font-size: ${theme.fonts.sizes.small};
    color: ${theme.colors.primary};
    text-decoration: none;
    position: relative;

    &::after {
      content: '';
      position: absolute;
      background: ${theme.colors.secondary};
      left: 25%;
      bottom: 0;
      width: 50%;
      height: 3px;
      transform: scale(0);
      transition: all 0.4s ease-in-out;
    }

    &:hover::after {
      transform: scale(1);
    }
  `}
`;
