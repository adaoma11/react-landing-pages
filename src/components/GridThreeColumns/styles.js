import styled from 'styled-components';
import { Title as GridItemTitle } from '../Heading/styles';

export const GridContainer = styled.div`
  display: grid;
  grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
  counter-reset: grid-counter;
  width: 100%;
  gap: 50px;
`;

export const GridItem = styled.div`
  margin-top: 80px;

  ${GridItemTitle} {
    position: relative;
    left: 4rem;
    margin-bottom: 30px;
  }

  ${GridItemTitle}::before {
    counter-increment: grid-counter;
    content: counter(grid-counter);
    position: absolute;
    font-size: 5rem;
    top: -2rem;
    left: -4rem;
    transform: rotate(7deg);
  }
`;
