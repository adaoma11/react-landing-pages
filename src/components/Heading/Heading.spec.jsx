import { screen } from '@testing-library/react';
import { Heading } from '.';
import { renderTheme } from '../../styles/renderTheme';
import { theme } from '../../styles/theme';

describe('<Heading />', () => {
  it('should render the heading with default values', () => {
    renderTheme(<Heading title="titulo" subtitle="subtitulo" />);

    const title = screen.getByRole('heading', { name: /titulo/i });
    const subtitle = screen.getByText('subtitulo');

    expect(title).toHaveStyle({
      color: theme.colors.primary,
      'font-size': theme.fonts.sizes.xlarge,
      'text-transform': 'none',
    });

    expect(subtitle).toBeInTheDocument();
  });

  it('should render with light color', () => {
    renderTheme(<Heading dark={false} title="texto" />);
    const title = screen.getByRole('heading', { name: /texto/i });
    expect(title).toHaveStyle({ color: theme.colors.white });
  });

  Object.keys(theme.fonts.sizes).forEach((size) => {
    it(`should render with ${size} text-size`, () => {
      renderTheme(<Heading size={size} title="texto" />);
      const title = screen.getByRole('heading', { name: 'texto' });
      expect(title).toHaveStyle({ 'font-size': theme.fonts.sizes[size] });
    });
  });

  it(`should render with correct font-size on small screens ${theme.medias.small}`, () => {
    renderTheme(<Heading size="xlarge" title="texto" />);
    const title = screen.getByRole('heading', { name: 'texto' });
    expect(title).toHaveStyleRule('font-size', theme.fonts.sizes.large, {
      media: '(max-width:  768px)',
    });
  });

  it('should render with uppercase', () => {
    renderTheme(<Heading upperCase title="texto" />);
    const title = screen.getByRole('heading', { name: 'texto' });
    expect(title).toHaveStyle({
      'text-transform': 'uppercase',
    });
  });

  it('should render with the proper title tag', () => {
    renderTheme(<Heading as="h6" title="texto" />);
    const title = screen.getByRole('heading', { name: 'texto' });
    expect(title.tagName.toLowerCase()).toBe('h6');
  });

  it('should match snapshot', () => {
    const { container } = renderTheme(<Heading title="Texto" />);
    expect(container).toMatchSnapshot();
  });
});
