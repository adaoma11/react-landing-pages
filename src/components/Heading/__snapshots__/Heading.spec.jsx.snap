// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Heading /> should match snapshot 1`] = `
.c0 {
  margin-bottom: 30px;
}

.c1 {
  color: #0A1128;
  font-size: 3rem;
  margin-bottom: 1rem;
  text-transform: none;
}

@media (max-width: 768px) {
  .c1 {
    font-size: 2rem;
  }
}

<div>
  <header
    class="c0"
  >
    <h1
      class="c1"
    >
      Texto
    </h1>
  </header>
</div>
`;
